exporter/jaeger/jaeger.go

```diff
@@ -221,6 +223,14 @@ func attributeToTag(key string, a interface{}) *gen.Tag {
 	return tag
 }
 
+// convertTime returns startTime/duration in microseconds.
+func convertTime(st, et time.Time) (startTime, duration int64) {
+	startTime = st.UnixNano() / 1000
+	endTime := et.UnixNano() / 1000
+	duration = endTime - startTime
```

Comment:
Why not compute the duration explicitly: 
```
duration = endTime.Sub(startTime) / time.Microsecond
```

This will use the monotonic clock. I understand that the EndTime should already be adjusted to be in the reference frame of the StartTime so this should always be the same but it is easier if the reader of this code doesn't need to know that and can locally determine that its correct.

Reply:
That is just exact the same as the existing code:
```
Duration:      data.EndTime.Sub(data.StartTime).Nanoseconds() / 1000,
```
because
```
// Nanoseconds returns the duration as an integer nanosecond count.
func (d Duration) Nanoseconds() int64 { return int64(d) }
```
and
```
const (
        Nanosecond  Duration = 1
        Microsecond          = 1000 * Nanosecond
)
```

Reply:
As I explained before, the existing code will output wrong data (ChildSpan.End>ParentSpan.End) to Jaeger under some circumstances, and Jaeger will adjusted these data due to "**child not fits within the parent span**", the result UI looks very bad.

A good trace:
<img width="747" alt="2018-07-26 10 04 54" src="https://user-images.githubusercontent.com/1009445/43237543-e3d87578-90bc-11e8-83eb-a71f0445188d.png">

and a **BAD** one:
<img width="757" alt="2018-07-26 10 03 55" src="https://user-images.githubusercontent.com/1009445/43237564-f93e329a-90bc-11e8-89d4-1be17c9e1132.png">

Reply:
Look at the existing code:
```
StartTime:     data.StartTime.UnixNano() / 1000,
Duration:      data.EndTime.Sub(data.StartTime).Nanoseconds() / 1000,
```

the problem here is:
1. StartTime will discards hundreds of nanoseconds.
2. data.EndTime.Sub(data.StartTime) will use those hundreds of nanoseconds in calculation.

Reply:
Yes this is a tricky one and only happens if the end time between child and parent span is in the sub microsecond range so rounding differences can make this occur.

Your fix solves the issue of a child span ending after a parent span due to these rounding differences but it comes at the expense of the durations being less accurate as they could be.

I personally favor duration accuracy over span end time ordering especially since a backend could remedy this while a duration inaccuracy can't be resolved. But then again how much info does a rounding error of 1 microsecond really mean.

That said, I'm ok with the proposed change.

Reply:
Oh I think I misunderstood the problem. Looking at what the Jaeger Go client does, it seems to do the same thing as our previous code:

In BuildJaegerThrift:
```go
duration := span.duration.Nanoseconds() / int64(time.Microsecond)
```

In Span.FinishWithOptions:
```go
s.duration = options.FinishTime.Sub(s.startTime)
```

So I actually think we should leave it as it was before since that's the behavior of the official Jaeger client.